control-plane/api/v1alpha1/jwtprovider_webhook.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package v1alpha1
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/consul-k8s/control-plane/api/common"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:object:generate=false
+
+type JWTProviderWebhook struct {
+	Logger logr.Logger
+
+	// ConsulMeta contains metadata specific to the Consul installation.
+	ConsulMeta common.ConsulMeta
+
+	decoder *admission.Decoder
+	client.Client
+}
+
+// NOTE: The path value in the below line is the path to the webhook.
+// If it is updated, run code-gen, update subcommand/controller/command.go
+// and the consul-helm value for the path to the webhook.
+//
+// NOTE: The below line cannot be combined with any other comment. If it is it will break the code generation.
+//
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-v1alpha1-jwtprovider,mutating=true,failurePolicy=fail,groups=consul.hashicorp.com,resources=jwtproviders,versions=v1alpha1,name=mutate-jwtprovider.consul.hashicorp.com,sideEffects=None,admissionReviewVersions=v1beta1;v1
```

Comment:
```suggestion
// +kubebuilder:webhook:verbs=create;update,path=/mutate-v1alpha1-jwtproviders,mutating=true,failurePolicy=fail,groups=consul.hashicorp.com,resources=jwtproviders,versions=v1alpha1,name=mutate-jwtprovider.consul.hashicorp.com,sideEffects=None,admissionReviewVersions=v1beta1;v1
```

Wondering if this should be plural: (samenessgroup and others have this as plural)

Reply:
I'm still wondering...

Ingress gateway is not plural:

https://github.com/hashicorp/consul-k8s/blob/2261a1b5aa4d44516e7b5722527d59d4c23875f5/control-plane/api/v1alpha1/ingressgateway_webhook.go#L34

Mesh config entry is not plural:

https://github.com/hashicorp/consul-k8s/blob/2261a1b5aa4d44516e7b5722527d59d4c23875f5/control-plane/api/v1alpha1/mesh_webhook.go

Service resolver is not plural:

https://github.com/hashicorp/consul-k8s/blob/2261a1b5aa4d44516e7b5722527d59d4c23875f5/control-plane/api/v1alpha1/serviceresolver_webhook.go#L34

Service router is not plural:

https://github.com/hashicorp/consul-k8s/blob/2261a1b5aa4d44516e7b5722527d59d4c23875f5/control-plane/api/v1alpha1/servicerouter_webhook.go#L34

Things like proxy-defaults and service-defaults and service-intentions are plural, but those config entry names were already plural where as jwt-provider is singular.